From 41d82f4dc4473803d1447a0eac524112bed032d1 Mon Sep 17 00:00:00 2001
From: Rui Miguel Silva <rui.silva@linaro.org>
Date: Tue, 27 Jul 2021 23:34:57 +0100
Subject: [PATCH] corstone1000: enable isp1763 and usb stack

Upstream-Status: Pending [Not submitted to upstream yet]
Signed-off-by: Arpita S.K <arpita.s.k@arm.com>

MPS3 board have a ISP1763 usb controller, add the correspondent mmio
area and enable it to be used for mass storage access for example.

Signed-off-by: Rui Miguel Silva <rui.silva@linaro.org>

---
 board/armltd/corstone1000/corstone1000.c |  8 ++++++++
 configs/corstone1000_defconfig           | 18 ++++++++----------
 2 files changed, 16 insertions(+), 10 deletions(-)

diff --git a/board/armltd/corstone1000/corstone1000.c b/board/armltd/corstone1000/corstone1000.c
index ab98fa87fbcd..a51f5ddfa002 100644
--- a/board/armltd/corstone1000/corstone1000.c
+++ b/board/armltd/corstone1000/corstone1000.c
@@ -46,6 +46,14 @@ static struct mm_region corstone1000_mem_map[] = {
 		.attrs = PTE_BLOCK_MEMTYPE(MT_DEVICE_NGNRNE) |
 			 PTE_BLOCK_NON_SHARE |
 			 PTE_BLOCK_PXN | PTE_BLOCK_UXN
+        }, {
+                /* USB */
+                .virt = 0x40200000UL,
+                .phys = 0x40200000UL,
+                .size = 0x00100000UL,
+                .attrs = PTE_BLOCK_MEMTYPE(MT_DEVICE_NGNRNE) |
+                         PTE_BLOCK_NON_SHARE |
+                         PTE_BLOCK_PXN | PTE_BLOCK_UXN
 	}, {
 		/* OCVM */
 		.virt = 0x80000000UL,
diff --git a/configs/corstone1000_defconfig b/configs/corstone1000_defconfig
index 54c746d829cb..8c0329611ba8 100644
--- a/configs/corstone1000_defconfig
+++ b/configs/corstone1000_defconfig
@@ -1,13 +1,12 @@
 CONFIG_ARM=y
-CONFIG_ARM64=y
 CONFIG_TARGET_CORSTONE1000=y
 CONFIG_SYS_TEXT_BASE=0x02100000
 CONFIG_SYS_MALLOC_F_LEN=0x2000
 CONFIG_NR_DRAM_BANKS=1
 CONFIG_IDENT_STRING=" corstone1000 aarch64 "
-CONFIG_BOOTDELAY=3
-CONFIG_OF_CONTROL=y
 CONFIG_DEFAULT_DEVICE_TREE="corstone1000-mps3"
+CONFIG_FIT=y
+CONFIG_BOOTDELAY=3
 CONFIG_USE_BOOTARGS=y
 CONFIG_BOOTARGS="console=ttyAMA0 loglevel=9"
 # CONFIG_DISPLAY_CPUINFO is not set
@@ -16,12 +15,11 @@ CONFIG_HUSH_PARSER=y
 CONFIG_SYS_PROMPT="corstone1000# "
 # CONFIG_CMD_CONSOLE is not set
 CONFIG_CMD_BOOTZ=y
-CONFIG_CMD_BOOTM=y
 # CONFIG_CMD_XIMG is not set
 # CONFIG_CMD_EDITENV is not set
 # CONFIG_CMD_ENV_EXISTS is not set
-# CONFIG_CMD_ARMFLASH=y
 # CONFIG_CMD_LOADS is not set
+CONFIG_CMD_USB=y
 # CONFIG_CMD_ITEST is not set
 # CONFIG_CMD_SETEXPR is not set
 CONFIG_CMD_DHCP=y
@@ -29,12 +27,12 @@ CONFIG_CMD_DHCP=y
 CONFIG_CMD_MII=y
 CONFIG_CMD_PING=y
 CONFIG_CMD_CACHE=y
-CONFIG_FIT=y
-# CONFIG_CMD_MISC is not set
 CONFIG_CMD_FAT=y
-CONFIG_DM=y
+CONFIG_OF_CONTROL=y
+CONFIG_REGMAP=y
 # CONFIG_MMC is not set
-# CONFIG_MTD_NOR_FLASH=y
 CONFIG_DM_ETH=y
 CONFIG_DM_SERIAL=y
-CONFIG_OF_LIBFDT=y
+CONFIG_USB=y
+CONFIG_USB_ISP1760=y
+CONFIG_USB_STORAGE=y
